### ISSUE TITLE
Keep all tensors on device between LLM decode iterations

### ISSUE THREAD

**Issue by odjuricicTT** [tenstorrent/tt-xla#3617]
## Summary

During LLM decode, every iteration round-trips tensors through the CPU host. This is a significant performance bottleneck, especially for multi-chip setups where device-to-host transfers are more expensive. Currently, transferring the output tensors to host alone adds ~20ms of overhead per decode step. We should keep all intermediate tensors on device between decode steps by wrapping the LLM in a PyTorch model that performs the post-processing (argmax, cache position update) on-device.

## Current Behavior

In [`generate_and_benchmark`](tests/benchmark/benchmarks/llm_benchmark.py#L182-L255), each decode iteration does:

1. **Forward pass** on device → produces `output` (logits)
2. **Transfer logits to CPU**: `logits = read_logits_fn(output).to("cpu")` (line 218) — **~20ms overhead**
3. **Argmax on CPU**: `next_token_ids = logits[:, -1].argmax(dim=-1)` (line 220)
4. **Transfer next_token_ids back to device**: `next_token_ids.unsqueeze(-1).to(device)` (line 237)
5. **Transfer cache_position to CPU and back**: lines 239-241

This means every single decode step involves **two device↔host round-trips** (logits down, input_ids + cache_position back up), which adds significant latency per token. The output tensor transfer alone (step 2) costs ~20ms, which is a substantial portion of the per-token decode time.

## Proposed Solution

Create a wrapper `nn.Module` around the LLM that includes the post-processing logic (argmax, cache_position increment) as part of the compiled graph, so everything stays on device:

```python
class LLMDecodeWrapper(nn.Module):
    def __init__(self, model, read_logits_fn):
        super().__init__()
        self.model = model
        self.read_logits_fn = read_logits_fn

    def forward(self, input_ids, past_key_values, cache_position, use_cache=True):
        output = self.model(
            input_ids=input_ids,
            past_key_values=past_key_values,
            cache_position=cache_position,
            use_cache=use_cache,
        )
        logits = self.read_logits_fn(output)
        next_token_ids = logits[:, -1].argmax(dim=-1).unsqueeze(-1)
        next_cache_position = cache_position[-1:] + 1
        # Return both the token ids and updated cache position
        # so the next iteration can consume them directly on device
        return next_token_ids, next_cache_position, logits
```

### Key considerations

- **Additional outputs**: The wrapper needs to return `next_token_ids` and `next_cache_position` as additional outputs alongside logits. This changes the compiled graph signature.
- **EOS checking**: Currently EOS is checked every iteration on CPU. With on-device tensors, we'd either check less frequently (e.g. every N steps) or compile the EOS check into the graph as well.
- **PCC validation**: Logits still need to come back to CPU for PCC comparison, but only for the first iteration (validation), not every decode step.
- **Compilation**: The wrapper must be compatible with `torch.compile(backend="tt")`. The argmax and cache_position arithmetic are simple ops that should compile cleanly.
- **Multi-chip sharding**: For tensor-parallel setups, the wrapper outputs need proper sharding marks. The `next_token_ids` and `next_cache_position` should be replicated across devices.
